src/cli/cli.cc

```diff
@@ -184,11 +237,7 @@ StatusCode CheckIn(AkliteClientExt &client, const LocalUpdateSource *local_updat
       std::cout << "\n";
     }
   }
-  if (!cr) {
-    return res2StatusCode<CheckInResult::Status>(c2s, cr.status);
-  }
 
-  auto gti_res = client.GetTargetToInstall(cr);
   if (gti_res && gti_res.selected_target.IsUnknown()) {
```

Comment:
It looks like this is needed only for the json type of output?

Reply:
It is not related to the json output. This was added here: https://github.com/foundriesio/aktualizr-lite/pull/345/commits/487ef43eead80f51855691aae59705192431659f with the objective of allowing the CLI check command result to be used directly to decide if a pull/install is expected to be called. I.e., if there is a target that need to be installed.

Reply:
OK. Do we need it for the `list` too?

Reply:
I would keep it for the `list` operation as well. There is no change of state, and the adequate return code might be useful if someone is trying to use the currently stored metadata to decide on doing or not an update.

Reply:
ok.